tests/Unit/Evolution/Test_TagsDomain.cpp

```diff
@@ -0,0 +1,269 @@
+      domain::Tags::FunctionsOfTime,
+      domain::CoordinateMaps::Tags::CoordinateMap<Dim, Frame::Grid,
+                                                  Frame::Inertial>>;
+  using compute_tags = db::AddComputeTags<
+      domain::Tags::CoordinatesMeshVelocityAndJacobiansCompute<
+          domain::CoordinateMaps::Tags::CoordinateMap<Dim, Frame::Grid,
+                                                      Frame::Inertial>>,
+      domain::Tags::InertialFromGridCoordinatesCompute<Dim>,
+      domain::Tags::ElementToInertialInverseJacobian<Dim>,
+      domain::Tags::InertialMeshVelocityCompute<Dim>,
+      evolution::domain::Tags::DivMeshVelocityCompute<Dim>>;
+
+  MAKE_GENERATOR(gen);
+  const std::array<double, 3> velocity{{1.2, 0.2, -8.9}};
+  const double initial_time = 0.0;
+  const std::array<std::string, 3> functions_of_time_names{
+      {"TranslationX", "TranslationY", "TranslationZ"}};
+
+  UniformCustomDistribution<double> dist(-10.0, 10.0);
+
+  const Mesh<Dim> mesh{5, Spectral::Basis::Legendre,
+                       Spectral::Quadrature::GaussLobatto};
+  const size_t num_pts = mesh.number_of_grid_points();
+
+  tnsr::I<DataVector, Dim, Frame::Grid> grid_coords{num_pts};
+  fill_with_random_values(make_not_null(&grid_coords), make_not_null(&gen),
+                          make_not_null(&dist));
+  InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Grid>
+      element_to_grid_inverse_jacobian{num_pts};
+  fill_with_random_values(make_not_null(&element_to_grid_inverse_jacobian),
+                          make_not_null(&gen), make_not_null(&dist));
+
+  std::unordered_map<std::string,
+                     std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+      functions_of_time{};
+  functions_of_time[functions_of_time_names[0]] =
+      std::make_unique<domain::FunctionsOfTime::PiecewisePolynomial<2>>(
+          initial_time,
+          std::array<DataVector, 3>{{{0.0}, {velocity[0]}, {0.0}}});
+  functions_of_time[functions_of_time_names[1]] =
+      std::make_unique<domain::FunctionsOfTime::PiecewisePolynomial<2>>(
+          initial_time,
+          std::array<DataVector, 3>{{{0.0}, {velocity[1]}, {0.0}}});
+  functions_of_time[functions_of_time_names[2]] =
+      std::make_unique<domain::FunctionsOfTime::PiecewisePolynomial<2>>(
+          initial_time,
+          std::array<DataVector, 3>{{{0.0}, {velocity[2]}, {0.0}}});
+
+  const std::unique_ptr<
+      domain::CoordinateMapBase<Frame::Grid, Frame::Inertial, Dim>>
+      grid_to_inertial_map = std::make_unique<
+          domain::CoordinateMap<Frame::Grid, Frame::Inertial,
+                                domain::CoordinateMaps::Identity<Dim>>>();
+
+  const double time = 3.0;
+  auto box = db::create<simple_tags, compute_tags>(
+      time, mesh, grid_coords, element_to_grid_inverse_jacobian,
+      std::move(functions_of_time), grid_to_inertial_map->get_clone());
+
+  CHECK_FALSE(static_cast<bool>(db::get<domain::Tags::DivMeshVelocity>(box)));
```

Comment:
For my understanding: is the idea that in a time-independent context, the time-dependent stuff produces an invalid boost:optional?